Assets/Scripts/Gameplay/TurnController.cs

```diff
@@ -0,0 +1,47 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Gameplay
+{
+    public class TurnController : MonoBehaviour
+    {
+        [SerializeField]
+        public int currentTurn = 0;
+
+        //private Dictionary<int, Player> playerList;
+        private List<Player> turnOrder = new();
+
+        public void AddPlayer(Player player)
+        {
+            if (turnOrder.Contains(player))
+                return;
+
+            turnOrder.Add(player);
+        }
+
+        public void ResetTurnOrder()
+        {
+            turnOrder = (List<Player>)turnOrder.Shuffle();
+            currentTurn = 0;
+        }
+
+        public Player CurrentPlayer()
+        {
+            if (turnOrder.Count < 1)
+            {
+                return null;
+            }
+
+            return turnOrder[currentTurn];
+        }
+
+        public void NextTurn()
+        {
+            currentTurn++;
```

Comment:
nit: I think this can be simplified to `currentTurn = currentTurn % turnOrder.Count`